master_server/master_server.py

```diff
@@ -0,0 +1,269 @@
+    self.request_status = request_status
+    self.last_vm_assigned = last_vm_assigned
+    self.number_of_retries_allowed = number_of_retries_allowed
+    self.number_of_retries_till_now = number_of_retries_till_now
+
+def create_new_request(
+    request_proto_file, request_status, last_vm_assigned,
+    number_of_retries_allowed, number_of_retries_till_now):
+  """This function adds a new row to database.
+  """
+  dessert = Newrequestentry(
+      request_proto_file, request_status, last_vm_assigned,
+      number_of_retries_allowed, number_of_retries_till_now)
+  db.session.add(dessert)
+  db.session.commit()
+  return dessert
+
+def get_status(request_id):
+  """Get status of previous request.
+  """
+  get_row_data = Newrequestentry.query.get(request_id)
+  return get_row_data
+
+def change_state(request_id, input_file):
+  """Change request_status of a request.
+  """
+  Newrequestentry.query.filter_by(request_id=request_id).\
+  update({Newrequestentry.response_proto_file: input_file})
+  db.session.commit()
+
+def change_state_again(request_id, last_vm_assigned,
+      number_of_retries_till_now):
+  """Change request_status of a request.
+  """
+  Newrequestentry.query.filter_by(request_id=request_id).\
+  update({Newrequestentry.last_vm_assigned: last_vm_assigned, Newrequestentry.\
+      number_of_retries_till_now: number_of_retries_till_now})
+  db.session.commit()
+
+@app.route('/', methods=['GET', 'POST'])
+def hello_world():
+  return(' You have reached master server \n')
+
+@app.route('/register', methods=['GET', 'POST'])
+def add_vm_address():
+  """Add a VM to working_vm_address_list list.
+  """
+  address = request.data.decode('utf-8')
+  if address not in working_vm_address_list:
+    working_vm_address_list.append(address)
+  return {'message': 'success'}
+
+@app.route('/upload')
+def upload_file():
+  """Serve web page
+  """
+  return render_template('upload.html')
+
+@app.route('/uploader', methods=['GET', 'POST'])
+def upload_fil():
+  """
+    Accept proto file from user and
+    assign it to VM.
+  """
+  f = request.files['file']
+  read = f.read()
+  vm_n = find_working_vm()
+  if vm_n == 'NOT':
+    return 'try again later'
+  else:
+    try:
+      working_vm_address_list.remove(vm_n)
+      task_request = Request_pb2.TaskRequest()
+      task_request.ParseFromString(read)
+      requ = create_new_request(
+          read, 'accepted', vm_n, task_request. number_of_retries, 0)
+      task_request.request_id = requ.request_id
+      response = requests.post(url=vm_n, files={'task_request':
+          task_request.SerializeToString()})
+      return 'file uploaded successfully,your request_id is '+ \
+          str(requ.request_id)+'  '+' address is '+str(requ.last_vm_assigned)+ \
+          ' number of retries allowed is '+str(requ.number_of_retries_allowed)
+    except:
+      return 'try again later'
+
+@app.route('/vm_status', methods=['GET', 'POST'])
+def upload_fi():
```

Comment:
upload_file*

Reply:
I see that is already used, please use more descriptive function names.

Reply:
Changes have been made as suggested.